tests/test_torch_wl_kernel.py

```diff
@@ -26,157 +29,199 @@ def example_graphs(self):
 
         return [G1, G2, G3]
 
-    @pytest.mark.parametrize("n_iter", [1, 2, 3, 4, 5, 6, 7])
-    @pytest.mark.parametrize("normalize", [True, False])
+    def create_adjacency_and_labels(self, graphs):
+        """Helper method to create adjacency matrices and label tensors from graphs."""
+        adjacency_matrices = []
+        label_tensors = []
+
+        # Create a consistent label mapping
+        label_dict = {}
+        current_label = 0
+
+        for graph in graphs:
+            # Create adjacency matrix
+            adj = nx.adjacency_matrix(graph).tocoo()
+            indices = torch.LongTensor([adj.row, adj.col])
+            values = torch.FloatTensor(adj.data)
+            size = torch.Size([graph.number_of_nodes(), graph.number_of_nodes()])
+            sparse_adj = torch.sparse_coo_tensor(
+                indices, values, size, device=self.device
+            ).to_sparse_csr()
+            adjacency_matrices.append(sparse_adj)
+
+            # Create label tensor with controlled mapping
+            node_labels = []
+            for node in range(graph.number_of_nodes()):
+                if "label" in graph.nodes[node]:
+                    label = graph.nodes[node]["label"]
+                    if label not in label_dict:
+                        label_dict[label] = current_label
+                        current_label += 1
+                    node_labels.append(label_dict[label])
+                else:
+                    node_labels.append(node)
+
+            label_tensors.append(
+                torch.tensor(node_labels, dtype=torch.long, device=self.device)
+            )
+
+        return adjacency_matrices, label_tensors
```

Comment:
We'll likely need this functionality in the main src code, not just in the tests